File: check_sitemap.js
import https from 'https';

const urls = [
    'https://yousef-bakr-s-portfolio.vercel.app/sitemap.xml',
    'https://yousef-bakr-s-portfolio.vercel.app/robots.txt'
];

function checkUrl(index) {
    if (index >= urls.length) return;
    const url = urls[index];
    https.get(url, (res) => {
        console.log(`\nURL: ${url}`);
        console.log(`Status: ${res.statusCode}`);
        console.log(`Content-Type: ${res.headers['content-type']}`);
        console.log(`X-Robots-Tag: ${res.headers['x-robots-tag'] || 'Not Set'}`);
        res.resume(); // Consume response to free memory
        res.on('end', () => checkUrl(index + 1));
    }).on('error', (e) => {
        console.error(`Error fetching ${url}:`, e);
        checkUrl(index + 1);
    });
}

checkUrl(0);

File: check_sitemap.py
import requests

url = "https://yousef-bakr-s-portfolio.vercel.app/sitemap.xml"
try:
    response = requests.get(url)
    print(f"Status Code: {response.status_code}")
    print("Headers:")
    for key, value in response.headers.items():
        print(f"{key}: {value}")
    print(f"\nContent Preview:\n{response.text[:200]}")
except Exception as e:
    print(f"Error: {e}")
